### PR TITLE
Do not close cookie list opt-in bubble on deactivate

### PR DESCRIPTION
**Pull request by zenparsing**
Resolves https://github.com/brave/brave-browser/issues/25693

## Submitter Checklist:

- [ ] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [ ] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [ ] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [ ] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [ ] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [x] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [x] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [x] Major classes, functions and non-trivial code blocks are well-commented
- [x] Changes in component dependencies are properly reflected in `gn`
- [x] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [x] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### CODE REVIEW

**Review by sangwoo108 — APPROVED** [review]
LGTM! I didn't know that [WebUIBubbleManagerT](https://source.chromium.org/chromium/chromium/src/+/main:chrome/browser/ui/views/bubble/webui_bubble_manager.h;bpv=1;bpt=1;l=97?q=WebUIBubbleManagerT&ss=chromium&gsn=WebUIBubbleManagerT&gs=kythe%3A%2F%2Fchromium.googlesource.com%2Fchromium%2Fsrc%3Flang%3Dc%252B%252B%3Fpath%3Dchrome%2Fbrowser%2Fui%2Fviews%2Fbubble%2Fwebui_bubble_manager.h%23e_6bcB_QAPkoacLAH1KxYty4Bx2FSVWuNEaLTLQaLCA&gs=kythe%3A%2F%2Fchromium.googlesource.com%2Fchromium%2Fsrc%3Flang%3Dc%252B%252B%3Fpath%3Dchrome%2Fbrowser%2Fui%2Fviews%2Fbubble%2Fwebui_bubble_manager.h%23ERzLG81-LfFPjqUqeo9A4mi-e3CctuhGYWJC2XT0ksU&gs=kythe%3A%2F%2Fchromium.googlesource.com%2Fchromium%2Fsrc%3Flang%3Dc%252B%252B%3Fpath%3Dchrome%2Fbrowser%2Fui%2Fviews%2Fbubble%2Fwebui_bubble_manager.h%23hi9aUb14vHb5rSnNkKXx2qJkTegB7wzoEYCwytyfXqY) closes bubbles on tab activation! Lucky us!

**Comment by sangwoo108** [comment]
@simonhong I think this is what you mentioned in the beginning. Do you mind taking a look?

**Comment by simonhong** [comment]
> @simonhong I think this is what you mentioned in the beginning. Do you mind taking a look?

Yeah `set_close_on_deactivate()` what I thought :)

**Review by simonhong — APPROVED** [review]

File: browser/ui/views/brave_shields/cookie_list_opt_in_bubble_host.cc [hunk 1]
```diff
@@ -85,6 +85,25 @@ void ShowBubbleOnSessionRestore(base::WeakPtr<Browser> browser, Profile*, int) {
...
 }
 
+class BubbleManager : public WebUIBubbleManagerT<CookieListOptInUI> {
+ public:
+  BubbleManager(views::View* anchor_view, Profile* profile)
+      : WebUIBubbleManagerT<CookieListOptInUI>(anchor_view,
+                                               profile,
+                                               GURL(kCookieListOptInURL),
+                                               IDS_BRAVE_SHIELDS) {}
+
```
[the remark this answers is not in the document]
Comment (zenparsing) [reply browser/ui/views/brave_shields/cookie_list_opt_in_bubble_host.cc:95]: Done.

File: browser/ui/views/brave_shields/cookie_list_opt_in_bubble_host.cc [hunk 2]
```diff
@@ -85,6 +85,25 @@ void ShowBubbleOnSessionRestore(base::WeakPtr<Browser> browser, Profile*, int) {
...
+  base::WeakPtr<WebUIBubbleDialogView> CreateWebUIBubbleDialog(
+      const absl::optional<gfx::Rect>& anchor) override {
+    auto dialog_view =
+        WebUIBubbleManagerT<CookieListOptInUI>::CreateWebUIBubbleDialog(anchor);
+    if (dialog_view) {
+      dialog_view->set_close_on_deactivate(false);
+    }
+    return dialog_view;
```
[the remark this answers is not in the document]
Comment (zenparsing) [reply browser/ui/views/brave_shields/cookie_list_opt_in_bubble_host.cc:103]: Done.

File: browser/ui/views/brave_shields/cookie_list_opt_in_bubble_host.cc [hunk 3]
```diff
@@ -95,7 +114,7 @@ CookieListOptInBubbleHost::CookieListOptInBubbleHost(Browser* browser)
 CookieListOptInBubbleHost::~CookieListOptInBubbleHost() = default;
```
[the remark this answers is not in the document]
Comment (sangwoo108) [reply browser/ui/views/brave_shields/cookie_list_opt_in_bubble_host.cc:114]: Oh, I completely forgot about that 👍 thanks!

**Pull request merged**